lib/amazonka-core/src/Amazonka/Sign/V4.hs

```diff
@@ -42,7 +43,16 @@ presign ex rq a r ts = Base.signRequest meta mempty auth
         . pair (CI.original hAMZSignedHeaders) (toBS shs)
         . pair (CI.original hAMZToken) (toBS <$> _authSessionToken a)
 
-    digest = Base.Tag "UNSIGNED-PAYLOAD"
+    digest =
+      case _requestBody rq of
+        Chunked _ -> unsignedPayload
+        Hashed HashedStream {} -> unsignedPayload
```

Comment:
The first argument of both `HashedStream` and `HashedBytes` is a digest, which we should trust to avoid recomputation (or erroneously returning `unsignedPayload`). You can use https://hackage.haskell.org/package/memory-0.17.0/docs/Data-ByteArray-Encoding.html#v:convertToBase to achieve this.

Reply:
Done